optaplanner-core/src/main/java/org/optaplanner/core/impl/heuristic/selector/move/generic/SwapMoveSelectorFactory.java

```diff
@@ -59,8 +62,20 @@ public SwapMoveSelectorFactory(SwapMoveSelectorConfig moveSelectorConfig) {
                                 SelectionOrder.fromRandomSelectionBoolean(randomSelection));
         List<GenuineVariableDescriptor<Solution_>> variableDescriptorList =
                 deduceVariableDescriptorList(leftEntitySelector.getEntityDescriptor(), config.getVariableNameIncludeList());
-        return new SwapMoveSelector<>(leftEntitySelector, rightEntitySelector, variableDescriptorList,
-                randomSelection);
+        if (variableDescriptorList.size() == 1 && variableDescriptorList.get(0).isListVariable()) {
+            return new ListSwapMoveSelector<>(
+                    (ListVariableDescriptor<Solution_>) variableDescriptorList.get(0),
+                    leftEntitySelector,
+                    rightEntitySelector,
+                    randomSelection);
+        }
+        if (variableDescriptorList.stream().noneMatch(GenuineVariableDescriptor::isListVariable)) {
```

Comment:
(nitpick) noneMatch -> anyMatch and throw?

Reply:
I have tried that and found out the original approach is better because we already conditionally `return` in the block above. I find this structure easier to read:
```java
if (cond1) {
    return obj1;
}
if (cond2) {
    return obj2;
}
throw ex;
```
than
```java
if (cond1) {
    return obj1;
}
if (!cond2) {
    throw ex;
}
return obj2;
```